solver/src/settlement.rs

```diff
@@ -140,12 +140,34 @@ impl Settlement {
         match self.encoder.total_surplus(&external_prices) {
             Some(value) => value,
             None => {
-                tracing::error!("Overflow computing objective value for: {:?}", self);
+                tracing::error!("Overflow computing surplus value for: {:?}", self);
                 num::zero()
             }
         }
     }
 
+    // For now this computes the fees of all EOA trades.
+    pub fn total_fees(&self) -> U256 {
+        match self.encoder.total_fees() {
+            Some(value) => value,
+            None => {
+                tracing::error!("Overflow computing fees value for: {:?}", self);
+                0.into()
+            }
+        }
+    }
+
+    // For now this computes the fees of all EOA trades.
+    pub fn total_unsubsidized_fees(&self, fee_discount_factor: f64) -> U256 {
+        match self.encoder.total_unsubsidized_fees(fee_discount_factor) {
+            Some(value) => value,
+            None => {
+                tracing::error!("Overflow computing fees value for: {:?}", self);
+                0.into()
+            }
```

Comment:
nit: shouldn't the error handling logic be in the encoder? Or do you expect people calling encoder.total_unsubsidized_fees that don't care about the extra context?

Reply:
I was following along the existing logic around `total_surplus()` which defaults to zero for None. Do you prefer to keep it consistent or change this one to return an Option? If the latter, should we change total_surplus in a subsequent PR too?